common/browser_context.go

```diff
@@ -70,39 +74,82 @@ func (b *BrowserContext) AddCookies(cookies goja.Value) *goja.Promise {
 }
 
 // AddInitScript adds a script that will be initialized on all new pages.
-func (b *BrowserContext) AddInitScript(script goja.Value, arg goja.Value) {
+func (b *BrowserContext) AddInitScript(script goja.Value, arg goja.Value) error {
 	b.logger.Debugf("BrowserContext:AddInitScript", "bctxid:%v", b.id)
 
-	rt := b.vu.Runtime()
+	if !gojaValueExists(script) {
+		return errors.New("parsing init script: invalid script")
+	}
```

Comment:
Suggestion: We better put this in a previous refactoring commit first. All kinds of refactoring should be in a refactoring commit (or multiple ones) to make the new feature addition ready—as long as it makes sense. The reasoning is it makes it harder to see the actual changes (the goal of the PR)—the difference between the refactorings and the core changes.

_I know you did this because of my previous comment: I'm trying to educate you on how we work._